src/js/index.js

```diff
@@ -46,44 +47,58 @@ function handleDifficulty(difficultyChanged) {
 	return false;
 }
 
+function handleSort() {
+	Array.from(contentEl.children)
+		.map(child => contentEl.removeChild(child))
+		.sort(sort[sortingInput.value])
+		.forEach(sortedChild => contentEl.append(sortedChild));
+}
+
 function handleTags() {
 	const tags = selectr.getValue();
 
-	if (tags.length === 0) {
-		history.pushState(null, '', window.location.pathname);
-		return;
-	}
-
 	Array.from(contentEl.querySelectorAll('.item')).forEach(el => {
-		const show = tags.reduce((sho, tag) => sho || el.classList.contains(`tag-${tag}`), false);
+		const show = (showExpired.checked || !el.classList.contains('tag-expired')) &&
+			tags.reduce((sho, tag) => sho || el.classList.contains(`tag-${tag}`), tags.length === 0);
 		if (!show) {
 			el.classList.remove('visible');
 		}
 	});
 
-	if (!search) {
-		return;
-	}
-
 	search.set('tags', tags.join(' '));
-	const newRelativePathQuery = `${window.location.pathname}?${search.toString()}`;
-	history.pushState(null, '', newRelativePathQuery);
+
+	search.set('expired', showExpired.checked ? 'true' : '');
```

Comment:
```suggestion
	search.set('expired', showExpired.checked);
```

Reply:
I think @aslafy-z said we should remove the `expired` search param in case it is set to `false`, to keep the URL as short and simple as possible.
I think we could set `expired` to `false`, but not by default (only if the checkbox is deliberately deselected).

Reply:
It would get deleted by the updateUrl fn

`search.set('expired', false)`
`search.get('expired') => "false"`
`"false".trim() => "false"`
`!"false" => true`
  - Param is added to delete queue

Reply:
<img width="74" alt="Test" src="https://user-images.githubusercontent.com/22768990/51343944-3226bd80-1aa0-11e9-8e8b-4e1ae5fb4462.png">

Reply:
I dun goofed!

```suggestion
	search.set('expired', showExpired.checked || '');
```